src/h2_stream_set.erl

```diff
@@ -261,51 +349,160 @@ new_stream(
                     %% process, or it will just hang out there.
                     h2_stream:stop(Pid),
                     {error, ?REFUSED_STREAM, #closed_stream{id=StreamId}};
-                NewStreamSet ->
-                    {Pid, NewStreamSet}
+                ok ->
+                    {Pid, StreamId, StreamSet}
             end
     end.
 
+get_callback(#stream_set{callback_mod=CM, callback_opts = CO}) ->
+    {CM, CO}.
+
+socket(#stream_set{socket=Sock}) ->
+    Sock.
+
+connection(#stream_set{connection=Conn}) ->
+    Conn.
+
+get_self_settings(StreamSet) ->
+    try hd(ets:lookup(StreamSet#stream_set.table, self_settings)) of
+        #connection_settings{settings=locked} ->
+            timer:sleep(1),
```

Comment:
Need to do something about all the sleep's in his module, right?

Reply:
the goal was to force this process to yield on a conflict

Reply:
Oh. hmm